### DISCUSSION TITLE
Upload a large file

### DISCUSSION THREAD

**Discussion by raph124** [cockpit-project/cockpit#20157] [General]
Hey,
I would like to add a addon that allows you to upload a file. I use the file(...).replace(...) method to copy the file to my server. But every time I do this I get a timeout error because it is taking too long because the file is too large. Do you know if there is a solution to copy the file to my server without splitting it into parts, copying it and then reassembling it?
For example, a method to adjust the timeout? Because I didn't find anything like that in the documentation.
Thank you!

**Comment by jelly** [comment]
How big is your file upload? I've been working on file uploading for our file manager plugin https://github.com/jelly/cockpit-navigator/tree/file-upload My work is rather hackish, based on WiP cockpit-bridge chunking support. I haven't noticed any timeouts with this approach however.

cockpit.file.replace() should in theory send the data chunked https://github.com/cockpit-project/cockpit/blob/main/pkg/lib/cockpit.js#L3582 

But maybe you mean the timeout which closes the Cockpit session?  Do you have that configured in your cockpit.conf?

**Comment by raph124** [comment]
Hey, thank you for the reply.
Yes, I changed the cockpit.conf file but the problem still exists. My file upload is around 190 MB and part of the file is also being copied but then suddenly it stops and the error is as follows: 
![Error](https://github.com/cockpit-project/cockpit/assets/110560850/8ff6001a-0a30-48f4-bc0f-95fcd3a6e70f)

Here is also my function i used:
![Function](https://github.com/cockpit-project/cockpit/assets/110560850/4844937b-6d96-46e2-ac19-40747762337b)

What confuses me is that every time I try to upload the file, the entire file is copied to my server when I reload the page. but why am I getting the timeout error?

**Reply by jelly** [reply]
Interesting, the problem with this approach is that `replace` sends all data at once flooding the bridge. So my guess is that is why you get a health check error. We are working on a general better upload for navigator which should help with this case as well.
